### PR TITLE
vscode: Wrap with GAppsHook

### LINKED ISSUE #45286: vscode crash on 'open folder'
## Issue description
`GLib-GIO-ERROR **: 19:10:11.204: No GSettings schemas are installed on the system`
Seems to be related to #26560 and #38919

### Steps to reproduce
Try to open a folder with vscode.

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.18.1, NixOS, 18.09pre149415.8395f9aa85e (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixos, nixpkgs"`
 - channels(okina): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change
Fixes #45286

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by worldofpeace — APPROVED** [review]

File: pkgs/applications/editors/vscode/default.nix [hunk 1]
```diff
@@ -49,8 +49,8 @@ in
     };
 
     buildInputs = if stdenv.system == "x86_64-darwin"
-      then [ unzip makeWrapper libXScrnSaver libsecret ]
-      else [ makeWrapper libXScrnSaver libxkbfile libsecret ];
+      then [ unzip makeWrapper wrapGAppsHook libXScrnSaver libsecret ]
```
Comment (jtojnar) [inline pkgs/applications/editors/vscode/default.nix]: Darwin probably does not need it. If it is like Atom, they use a different front-end for MacOS.

**Comment by Chiiruno** [comment]
@jtojnar Okay, is this good?

**Review by jtojnar — COMMENTED** [review]
Ideally, `wrapGAppsHook` would go to `nativeBuildInputs` but electron packages are ugly abominations so it probably does not matter.

**Comment by Chiiruno** [comment]
@Mic92 All good?

**Comment by Mic92** [comment]
Is a backport required here?

**Comment by Chiiruno** [comment]
Backport?
What do you mean?

**Comment by Mic92** [comment]
is 18.03 also affected?

**Comment by Chiiruno** [comment]
I'm afraid I have no idea, but if the last commit I made to vscode, before this one, is on 18.03, then yes.

**Comment by gebner** [comment]
At least for me, #45286 only appeared after the upgrade vscode 1.26.0.  18.03 is on vscode 1.23.0, so it should be fine.  I think we only need to backport this fix if we also upgrade vscode on the release branch.

**Pull request merged**